client/src/useJobPaginator.test.js

```diff
@@ -0,0 +1,299 @@
+import { beforeEach, describe, expect, it, jest } from "@jest/globals";
+import { renderHook, act } from "@testing-library/react-hooks";
+import * as JobPaginatorState from "./JobPaginatorState";
+import AjaxHelper from "./AjaxHelper";
+import useJobPaginator from "./useJobPaginator";
+
+jest.mock("./AjaxHelper");
+jest.mock("./JobPaginatorState");
+
+describe("useJobPaginator", () => {
+    let ajax;
+
+    beforeEach(() => {
+        ajax = {
+            getImageUrl: jest.fn(),
+        };
+        window.alert = jest.fn();
+        window.confirm = jest.fn();
+        Object.defineProperty(window, "location", {
+            value: {
+                assign: jest.fn(),
+            },
+            writable: true,
+        });
+        AjaxHelper.getInstance.mockReturnValue(ajax);
+    });
+
+    it("increments and decrements the currentPage", async () => {
+        const { result } = await renderHook(() => useJobPaginator());
+
+        act(() => {
+            result.current.action.setOrder([
+                {
+                    filename: "test1",
+                    label: null,
+                },
+                {
+                    filename: "test2",
+                    label: null,
+                },
+            ]);
+        });
+
+        act(() => {
+            result.current.action.nextPage();
+        });
+
+        expect(result.current.state.currentPage).toBe(1);
+
+        act(() => {
+            result.current.action.prevPage();
+        });
+
+        expect(result.current.state.currentPage).toBe(0);
+    });
+
+    describe("setLabel", () => {
+        it("changes the label if null", () => {
+            const { result } = renderHook(() => useJobPaginator());
+
+            act(() => {
+                result.current.action.setOrder([
+                    {
+                        filename: "test1",
+                        label: null,
+                    },
+                ]);
+            });
+
+            act(() => {
+                result.current.action.setLabel(0, "test2");
+            });
+
+            expect(result.current.state.order[0].label).toEqual("test2");
+        });
+
+        it("does not change the label", () => {
```

Comment:
This is testing that set doesn't do anything if you specify an invalid index, correct? Maybe the description should incorporate that, like "does not change the label if index out of range".